lib/yerror.js

```diff
@@ -0,0 +1,11 @@
+const util = require('util')
```

Comment:
This was my only nitpick, thanks for bringing it up @satazor - here's the nodejs discussion on the discouragement of using `util.inherits`: https://github.com/nodejs/node/issues/4179